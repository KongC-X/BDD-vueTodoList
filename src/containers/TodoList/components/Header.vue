<template>
  <div>
    <input
        type="text"
        data-test="input"
        v-model="inputValue"
        @keyup.enter="addTodoItem"
    >
  </div>
</template>

<script>
export default {
  name: 'Header',
  props: {},
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style scoped lang="stylus">
</style>
