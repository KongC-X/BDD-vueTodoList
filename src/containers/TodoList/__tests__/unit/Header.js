import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import { findTestWrapper } from '@/utils/testUtil.js'

describe('Header 测试', () => {
  it('存在输入框', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    expect(input.exists()).toBe(true)
  })

  it('inputValue 为空，输入框为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('输入框内容变化，inputValue跟随变化', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('garen')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('garen')
  })

  it('输入框输入回车，无内容, 无反应', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('输入框输入回车，有内容, 向外发射事件, 并清空内容', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('garen')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    expect(wrapper.vm.$data.inputValue).toBe('')
  })

  it('UI快照', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })
})
