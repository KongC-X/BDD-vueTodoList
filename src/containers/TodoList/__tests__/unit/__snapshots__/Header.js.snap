// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 测试 UI快照 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text" data-test="input" placeholder="Add TodoItem" class="header-input"></div>
</div>
`;
